src/expr/src/scalar/like_pattern.rs

```diff
@@ -99,49 +99,60 @@ pub fn normalize_pattern(pattern: &str, escape: EscapeBehavior) -> Result<String
 // That said, regular expressions aren't that efficient. For most patterns
 // we can do better using built-in string matching.
 
-/// An object that can test whether a string matches a LIKE or ILIKE pattern.
-#[derive(Debug, Clone, Deserialize, Serialize, Derivative, MzReflect)]
-#[derivative(Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct Matcher {
-    pub pattern: String,
-    pub case_insensitive: bool,
-    #[derivative(
-        PartialEq = "ignore",
-        Hash = "ignore",
-        Ord = "ignore",
-        PartialOrd = "ignore"
-    )]
-    matcher_impl: MatcherImpl,
-}
+pub use matcher::Matcher;
+use matcher::MatcherImpl;
+
+// This lint interacts poorly with `derivative` here; we are confident it generates
+// compatible `PartialOrd` and `Ord` impls. Unfortunately it also requires we introduce
+// this module to allow it.
+#[allow(clippy::incorrect_partial_ord_impl_on_ord_type)]
```

Comment:
👎🏻 from me on contorting our code to let us turn on buggy new clippy lints. Can we just globally turn this one off instead? What's the value of keeping it on?

Reply:
Yeah, we might go ahead with that if this becomes too onerous, but there was only 1 example so I chose contortion